code/datums/components/anti_magic.dm

```diff
@@ -22,14 +23,16 @@
 	psychic = _psychic
 	if(_allowed_slots)
 		allowed_slots = _allowed_slots
+	if(_works_in_hands)
+		works_in_hands = _works_in_hands
 	if(!isnull(_charges))
 		charges = _charges
 	blocks_self = _blocks_self
 	reaction = _reaction
 	expire = _expire
 
 /datum/component/anti_magic/proc/on_equip(datum/source, mob/equipper, slot)
-	if(!CHECK_BITFIELD(allowed_slots, slotdefine2slotbit(slot)))
+	if(!(works_in_hands && slot == SLOT_HANDS) && !CHECK_BITFIELD(allowed_slots, slotdefine2slotbit(slot))) //If it's not handheld, check that the slot is valid for antimagic
```

Comment:
Why aren't you just adding SLOT_HANDS to allowed slots

Reply:
Now that i look at it again, i think i could just do that, although it would result in some awkward lines of code where i have to put in the whole allowed_slots list without hands as an argument in the add_component.

Reply:
Allowed slots isnt a list and you should probably make the default just be `ALL` tbh.
Let things pass in exactly what things they want if they only want specific slots

Reply:
It's mostly there to remove backpack slots, but doing it on negative still seems like a nice solution, i'll do it that way.